pandas/core/reshape/merge.py

```diff
@@ -1635,7 +1636,11 @@ def _get_merge_keys(self):
                 )
             )
 
-            if is_datetime64_dtype(lt) or is_datetime64tz_dtype(lt):
+            if(
+                is_datetime64_dtype(lt)
+                or is_datetime64tz_dtype(lt)
+                or is_timedelta64_dtype(lt)
+            ):
                 if not isinstance(self.tolerance, Timedelta):
```

Comment:
This should probably be `if not isinstance(self.tolerance, datetime.timedelta)` so users can pass `datetime.timedelta` objects. If you could add a test case for that situation that'd be great.

Reply:
see commit "mroeschke: this test is for you :rocket:"

Reply:
the test fails

Reply:
We'll need a different PR for this test I think, it fails at this conversion check. 
https://github.com/pandas-dev/pandas/blob/143bc34aa8f068b18e7137df7ca91b9929cc1389/pandas/core/reshape/merge.py#L1709

datetime.timedelta has no attribute "value"

Reply:
`datetime.timedelta` objects do not have nanoseconds like `pd.Timedelta` 
Another check and conversion is necessary to convert `datetime.timedelta` correctly to nanoseconds

Reply:
Okay thanks for investigating. Yes, I'd be great to address in another issues/PR.